pkg/route/engine.go

```diff
@@ -634,6 +634,20 @@ func (engine *Engine) recv(ctx *app.RequestContext) {
 	}
 }
 
+// HandleContext Re-enter a context that has been rewritten.
+// This can be done by setting ctx.Request.SetRequestURI() to new target and response will be reset.
+// Disclaimer: Maybe loop self to death with this, use wisely.
+func (engine *Engine) HandleContext(c context.Context, ctx *app.RequestContext) {
+	old := ctx.GetIndex()
+	ctx.Response.Reset()
+	ctx.Request.ResetURI()
+	ctx.Params = ctx.Params[:0]
+	ctx.Errors = ctx.Errors[0:0]
+	ctx.SetIndex(-1)
+	engine.ServeHTTP(c, ctx)
+	ctx.SetIndex(old)
```

Comment:
what is the purpose setting this back after .ServeHTTP() return?

Reply:
refer to https://github.com/gin-gonic/gin/pull/1764